bricks/test_optimizer/hooks/test/string_id_generator_test.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:hooks/pre_gen.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('$StringIdGenerator', () {
+    test('can be instantiated', () {
+      expect(StringIdGenerator.new, returnsNormally);
+    });
+
+    group('next', () {
+      test('returns normally', () {
+        final generator = StringIdGenerator();
+        expect(generator.next, returnsNormally);
+      });
+
+      test('generates unique strings', () {
```

Comment:
We should also test the expect order of the generated, what would happen after the last index was z? Will it be `zA` or `aa`?

Reply:
Do we care? My only concern is that they are unique.

Reply:
To ensure a certain uniqueness, yes ;-)

That is the goal of this class and the unit. Let's say this class will be used for something that is long lived, if this isn't tested and we change it we might accidentally introduce a breaking change because our output of the CLI would change.